features/topologies/service/api/src/main/java/org/opennms/netmgt/topologies/service/api/OnmsTopologyProtocol.java

```diff
@@ -44,6 +44,8 @@ public static OnmsTopologyProtocol allProtocols() {
     }
     
     final private String m_id;
+    private String m_name;
```

Comment:
Could these fields be made mandatory and set in the constructor? Given the design of the class right now these fields are effectively optional and it seems like they shouldn't be, although I'm not that familiar with this area.

Reply:
I would prefer not to. There are many changes to be done and actually the name is only used by the Service to set up the name in the topology Map.
We can make this changes in a later project.
Usually the ID will identify the protocol.....CDP, LLDP and so on....The name  should be optional.

Reply:
Ok, it seems to me that this type is being overloaded in this case... I'll leave it up to your judgement how to proceed since I don't know this area very well.